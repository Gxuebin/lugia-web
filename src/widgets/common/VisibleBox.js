/*
 * 可通过visible属性控制是否可见的盒子
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';

type VisibleBoxProps = {
  visible: boolean
}
export default styled.div`
   ${(props: VisibleBoxProps) => (props.visible ? '' : 'display: none;')}
`;
