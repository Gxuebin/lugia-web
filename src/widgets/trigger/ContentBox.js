import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Box = styled.div`
  display: ${props => (props.visible ? '' : 'none')};
`;

const MaskBox = Box.extend`
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: #373737;
  background-color: rgba(55, 55, 55, 0.6);
  height: 100%;
  filter: alpha(opacity=50);
  z-index: 1050;
`;

class ContentBox extends Component {
  static propTypes = {
    children: PropTypes.any,
    className: PropTypes.string,
    visible: PropTypes.bool,
    isMask: PropTypes.bool,
  };

  shouldComponentUpdate (nextProps) {
    return nextProps.visible !== this.props.visible;
  }

  render () {
    const { isMask, ...props } = this.props;
    if (isMask) {
      return <MaskBox {...this.props}/>;
    }
    if (React.Children.count(props.children) > 1) {
      return <Box {...this.props}/>;
    }

    return React.Children.only(props.children);
  }
}

export default ContentBox;
