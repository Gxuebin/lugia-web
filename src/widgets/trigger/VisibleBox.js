/*
 * 可通过visible属性控制是否可见的盒子
 */
import React from 'react';
import styled from 'styled-components';

export default styled.div`
  display: ${props => (props.visible ? '' : 'none')};
`;
