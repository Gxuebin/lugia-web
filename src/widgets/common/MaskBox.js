/*
 * 遮罩盒子
 * @flow
 */
import * as React from 'react';
import VisibleBox from '../common/VisibleBox';
import styled from 'styled-components';

export const ZIndex = 1050;
export default styled(VisibleBox)`
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: #373737;
  background-color: rgba(55, 55, 55, 0.6);
  height: 100%;
  filter: alpha(opacity=50);
  z-index: ${ZIndex};
`;
