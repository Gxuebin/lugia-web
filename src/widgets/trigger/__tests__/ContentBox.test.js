//@flow
import React from 'react';
import chai from 'chai';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import VisibleBox from '../../common/VisibleBox';
import MaskBox from '../../common/MaskBox';
import ContentBox from '../ContentBox';
import Enzyme, { mount, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const { expect: exp, } = chai;
Enzyme.configure({ adapter: new Adapter(), });

describe('ContentBox', () => {

  it('default children:onlyOne', () => {
    const onlySpan = renderer.create(<span>a</span>).toJSON();
    const contentBox = renderer.create(<ContentBox><span>a</span></ContentBox>).toJSON();
    expect(contentBox).toMatchObject(onlySpan);
    expect(contentBox).toMatchSnapshot();
  });

  it('default children: more', () => {
    const visibleBox = renderer.create(<VisibleBox><span>a</span><span>b</span></VisibleBox>).toJSON();
    const contentBox = renderer.create(<ContentBox><span>a</span><span>b</span></ContentBox>).toJSON();
    expect(contentBox).toMatchObject(visibleBox);
    expect(contentBox).toMatchSnapshot();
  });

  it('mask: false, visible: false, children: only one', () => {
    const onlySpan = renderer.create(<span>a</span>).toJSON();
    const contentBox = renderer.create(<ContentBox visible={false}><span>a</span></ContentBox>).toJSON();
    expect(contentBox).toMatchObject(onlySpan);
    expect(contentBox).toMatchSnapshot();

  });

  it('mask: true, visible: true, children: only one', () => {
    const maskBox = renderer.create(<MaskBox visible><span>a</span></MaskBox>).toJSON();
    const contentBox = renderer.create(<ContentBox visible isMask><span>a</span></ContentBox>).toJSON();
    expect(contentBox).toMatchObject(maskBox);
    expect(contentBox).toMatchSnapshot();

  });

  it('mask: false, children: more, visible: false', () => {
    const visibleBox = renderer.create(<VisibleBox visible={false}><span>a</span><span>b</span></VisibleBox>).toJSON();
    const contentBox = renderer.create(<ContentBox visible={false}><span>a</span><span>b</span></ContentBox>).toJSON();
    expect(contentBox).toMatchSnapshot();
    expect(contentBox).toMatchObject(visibleBox);
    expect(contentBox).toHaveStyleRule('display', 'none');
  });
  it('mask: true, children: more, visible: false', () => {
    const maskBox = renderer.create(<MaskBox visible={false}><span>a</span><span>b</span></MaskBox>).toJSON();
    const contentBox = renderer.create(<ContentBox isMask
                                                   visible={false}><span>a</span><span>b</span></ContentBox>).toJSON();
    expect(contentBox).toMatchSnapshot();
    expect(contentBox).toMatchObject(maskBox);
    expect(contentBox).toHaveStyleRule('display', 'none');
  });

  it('mask: true children: more', () => {
    const visibleBox = renderer.create(<VisibleBox visible={false}><span>a</span><span>b</span></VisibleBox>).toJSON();
    const contentBox = renderer.create(<ContentBox visible={false}><span>a</span><span>b</span></ContentBox>).toJSON();
    expect(contentBox).toMatchSnapshot();
    expect(contentBox).toMatchObject(visibleBox);
    expect(contentBox).toHaveStyleRule('display', 'none');
  });

  it('children变化时组件要更新视图', () => {

    const txt = '我心';
    const cmp = mount(<ContentBox>
      <div>{txt}</div>
    </ContentBox>);
    exp(cmp.text()).to.be.equal(txt);
    const textTwo = '全中国';
    cmp.setProps({ children: <div>{textTwo}</div>, });
    exp(cmp.text()).to.be.equal(textTwo);

  });
});
