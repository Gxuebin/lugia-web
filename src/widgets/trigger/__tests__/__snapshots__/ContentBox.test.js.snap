// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBox default children: more 1`] = `
.c0 {
  display: none;
}

<div
  className="c0"
>
  <span>
    a
  </span>
  <span>
    b
  </span>
</div>
`;

exports[`ContentBox default children:onlyOne 1`] = `
<span>
  a
</span>
`;

exports[`ContentBox mask: false, children: more, visible: false 1`] = `
.c0 {
  display: none;
}

<div
  className="c0"
>
  <span>
    a
  </span>
  <span>
    b
  </span>
</div>
`;

exports[`ContentBox mask: false, visible: false, children: only one 1`] = `
<span>
  a
</span>
`;

exports[`ContentBox mask: true children: more 1`] = `
.c0 {
  display: none;
}

<div
  className="c0"
>
  <span>
    a
  </span>
  <span>
    b
  </span>
</div>
`;

exports[`ContentBox mask: true, children: more, visible: false 1`] = `
.c0 {
  display: none;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: #373737;
  background-color: rgba(55,55,55,0.6);
  height: 100%;
  -webkit-filter: alpha(opacity=50);
  filter: alpha(opacity=50);
  z-index: 1050;
}

<div
  className="c0"
>
  <span>
    a
  </span>
  <span>
    b
  </span>
</div>
`;

exports[`ContentBox mask: true, visible: true, children: only one 1`] = `
.c0 {
  display: inherit;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: #373737;
  background-color: rgba(55,55,55,0.6);
  height: 100%;
  -webkit-filter: alpha(opacity=50);
  filter: alpha(opacity=50);
  z-index: 1050;
}

<div
  className="c0"
>
  <span>
    a
  </span>
</div>
`;
